tools/fragpipe/fragpipe.xml

```diff
@@ -0,0 +1,286 @@
+<tool id="fragpipe" name="FragPipe -  Academic Research and Education User License (Non-Commercial)" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" python_template_version="3.5">
+    <description>Data analysis for mass spectrometry-based proteomics</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
+
+    <stdio>
+        <!-- The error code from individual tools run by FragPipe is not always propogated to the FragPipe process exit code -->
+        <regex match="exit code: [^0]"
+               source="stdout"
+               level="fatal"
+               description="A FragPipe pipeline process returned non-zero exit code."/>
+    </stdio>
+
+    <version_command>fragpipe --help | grep 'FragPipe v'</version_command>
+
+    <!--
+        FragPipe wrapper command:
+        - Links input files
+        - Modifies database path in workflow file
+        - Runs Philosopher to add decoys and contaminants
+        - Runs FragPipe, passing paths to MSFragger, Philosopher, Python, and IonQuant
+    -->
+    <command detect_errors="exit_code"><![CDATA[
+        mkdir .cache .config tmp &&
+        export XDG_CONFIG_HOME=\$(pwd)/.config &&
+        export XDG_CACHE_HOME=\$(pwd)/.cache &&
+        export JAVA_OPTS="\$JAVA_OPTS -Xmx\$[ \${GALAXY_MEMORY_MB:-8192} / 1024 ]g -Djava.io.tmpdir=\$(pwd)/tmp" &&
+        export LD_LIBRARY_PATH="\$LD_LIBRARY_PATH:/usr/local/lib" &&
+        mkdir -p outputs &&
+        @CMD_IMPORTS@
+        @LINK_FRAGGER_DATABASE@
+        @LINK_SAMPLES@
+        @UPDATE_WORKFLOW@
+        #if not $workflow_only
+            #if str($database_options) != 'none'
+                philosopher workspace --init &&
+                '$__tool_directory__/genericize_db.py' ${database_name} genericized_db.fas &&
+                philosopher database --custom 'genericized_db.fas' --prefix rev_
+                #if str($database_options) != 'decoy_and_contam'
+                    --contam
+                #end if
+                --contamprefix &&
+                mv *-decoys-*.fas decoys-contam-database.fas &&
+            #end if
+            fragpipe --msfragger_key f28c4be2-759b0a11-6c1ea652-1240565a --ionquant_key 47cb2953-b4bf7726-2c96f5fe-8f1013aa --headless --threads $threads --ram \$[ \${GALAXY_MEMORY_MB:-8192} / 1024 ] --workflow fp.workflow --manifest fp.manifest --workdir outputs
+            --config-msfragger \${CONDA_PREFIX}/share/msfragger-*/MSFragger-*/MSFragger-*.jar --config-philosopher `which philosopher` --config-python `which python` --config-ionquant \${CONDA_PREFIX}/share/ionquant-*/IonQuant.jar
+            && cd outputs
+            && ln -s log_*.txt log.txt
+            #if $output_options and 'combined_outputs' in $output_options
+                && mkdir combined_outputs
+                && for f in combined*tsv; do [[ -e $\{f} || continue ]]; mv \${f} combined_outputs; done
```

Comment:
`-e $\{f}` looks wrong. Can you add a test covering this piece of code?

maybe a `find . -name "combined*tsv" -exec mv {} combined_outputs/\;`

Reply:
Basically this works to only `mv` the files if they exist but I agree it's too convoluted, I got tunnel-vision with the for-loop.

I think the line you suggested will still fail if there are no files, but something like `find . -name 'combined*tsv' | xargs -I {} mv {} combined_outputs` would work. Or just a bash conditional.

Reply:
I updated this to use the `xargs` command I quoted.